infra_event_notifier/backends/datadog.py

```diff
@@ -0,0 +1,39 @@
+import time
+import json
+
+import urllib.request
+
+from typing import Mapping
+
+
+def send_event(
+    title: str,
+    text: str,
+    tags: Mapping[str, str],
+    datadog_api_key: str,
+    alert_type: str,
+) -> None:
+    """
+    Sends an event to Datadog.
+
+    :param title: Title of DD event
+    :param text: Body of event
+    :param tags: dict storing event tags
+    :param datadog_api_key: DD API key for sending events
+    :param alert_type: Type of event if using an event monitor, see https://docs.datadoghq.com/api/latest/events/
+    """
+    # API docs: https://docs.datadoghq.com/api/latest/events/#post-an-event
+    payload = {
+        "title": title,
+        "text": text,
+        "tags": [f"{k}:{v}" for k, v in tags.items()],
+        "date_happened": int(time.time()),
+        "alert_type": alert_type,
+    }
+    json_data = json.dumps(payload)
+    data = json_data.encode("utf-8")
+    req = urllib.request.Request("https://api.datadoghq.com/api/v1/events", data=data)
+    req.add_header("DD-API-KEY", datadog_api_key)
+    req.add_header("Content-Type", "application/json; charset=utf-8")
+    with urllib.request.urlopen(req) as response:
+        json.loads(response.read().decode("utf-8"))
```

Comment:
You are not going to raise an exception if the return code is not a 200 here. 
You should. If the call fails with a valid HTTP500 response, urllib.request does not throw. 
urllib would throw for connection errors.

So you will have to check the return value:
`status_code = response.getcode()`

Reply:
I'm seeing that it does throw automatically on 50X errors, I used the same code but with `httpstat.us/502` as the url and it threw an `HTTPError`: 
```
Traceback (most recent call last):
  File "/Users/benmckerry/Documents/projects/infra-event-notifier/infra_event_notifier/backends/datadog.py", line 54, in <module>
    send_event(
  File "/Users/benmckerry/Documents/projects/infra-event-notifier/infra_event_notifier/backends/datadog.py", line 40, in send_event
    with urllib.request.urlopen(req) as response:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 502: Bad Gateway
```
same with a 400 error:
```
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```

Reply:
Added something that raises an exception when status > 202 to catch 30X responses as well